src/collectors/loggly/loggly.py

```diff
@@ -0,0 +1,246 @@
+# coding=utf-8
+
+"""
+Collect numeric value of a field from loggly search results.
+
+This collector will retrieve the values of a configured field from loggly API
+and publish each one on the log's specified timestamp.
+
+The collector will limit the query to a timeframe equal to the collector's
+duration back from now.
+
+#### Dependencies
+
+    * requests
+    * futures
+
+Example config file LogglyCollector.conf
+
+```
+enabled = true
+interval = 60
+subdomain = example
+username = loggly_user
+password = loggly_pass
+
+[ metric ]
+[[ test.metric ]]
+field = some.field
+query = tag:some_tag AND some_text_field='VALUE'
+```
+
+"""
+
+import diamond.collector as collector
+import concurrent.futures
+import configobj
+import datetime
+import requests
+import bunch
+import time
+import os
+
+from math import ceil, log10
+from diamond.metric import Metric
+from diamond.error import DiamondException
+
+
+def multi_getattr(obj, attr, **kw):
+    """Retrieve nested attributes."""
+    attributes = attr.split(".")
+    for i in attributes:
+        try:
+            obj = getattr(obj, i)
+            if callable(obj):
+                obj = obj()
+        except AttributeError:
+            if 'default' in kw:
+                return kw['default']
+            else:
+                raise
+    return obj
+
+
+class LogglyCollector(collector.Collector):
+
+    """Loggly Collector."""
+
+    def get_default_config_help(self):
+        """Update configuration help."""
+        config_help = super(
+            LogglyCollector, self).get_default_config_help()
+        config_help.update({
+            'id': "Metric identifier",
+            'field': "Name of the field to retrieve, required",
+            'count': "Whether to retrieve the count of matching events",
+            'timestamp_field': "Name of the timestamp field."
+                               "Defaults to 'timestamp'",
+            'timestamp_format': "Timestamp format as for strptime or 'int'"
+                                "for unix time. Defaults to 'int'",
+            'subdomain': "Loggly subdomain to retrieve results from",
+            'query': "Loggly search query",
+            'username': "Loggly username",
+            'password': "Loggly password"
+        })
+        return config_help
+
+    def process_config(self):
+        """Prepare metrics' configuration."""
+        def set_defaults(config):
+            defaults = {
+                'timestamp_field': 'timestamp',
+                'timestamp_format': 'int',
+                'field': 'count',
+                'count': False,
+                'query': '*'
+            }
+            [config.update({key: value}) for key, value in defaults.items()
+                if key not in config.keys()]
+
+        super(LogglyCollector, self).process_config()
+
+        self.config_bunch = bunch.bunchify(self.config)
```

Comment:
So I'd prefer not using the bunch package to parse configs. None of the other collectors use it, thus leading to an inconsistency.